flow/project.py

```diff
@@ -1520,6 +1521,12 @@ def run(self, jobs=None, names=None, pretend=False, np=None, timeout=None, num=N
         if num and num < 0:
             num = None
 
+        # The 'names' argument must be a sequence, not a string.
+        if isinstance(names, six.string_types):
+            raise ValueError(
+                "The names argument of FlowProject.run() must be a sequence of strings, "
+                "not a string.")
```

Comment:
What's the reason that we can't just handle this conversion internally?

Reply:
For the sake of a well-defined API. This could go either way...